### PR TITLE
A set of assorted tweaks and cleanups in the build system.

### PR DESCRIPTION
**Pull request by osctobe**

### CODE REVIEW

File: Makefile.config [hunk 1]
```diff
@@ -78,7 +78,7 @@ export DEFINES += $(FEATURE_DEFINES)
 export CFLAGS += $(FEATURE_DEFINES)
 
 FEATURES_LIST	:= TCP_REPAIR STRLCPY STRLCAT PTRACE_PEEKSIGINFO \
-	SETPROCTITLE_INIT MEMFD TCP_REPAIR_WINDOW MEMFD_CREATE \
```
Comment (0x7f454c46) [inline Makefile.config:81]: Fixes: 5ee1ac1f28e6 ("criu: remove FEATURE_TEST_MEMFD")

**Comment by 0x7f454c46** [comment]
`From: Uros Prestor <urosp@google.com>`
Should have SoB line that certifies that the author agrees with GPL2 license and all that legal staff.
Besides, I see your Gerrit's `Change-ID`s, which probably should be removed as they don't give anyone any context outside Google (or is your Gerrit publically visible?).
That's what we usually ask Virtuozzo/Parallels folks - to remove any BUG links that are not visible/accessible to CRIU contributors.

Other than this, LGTM

**Comment by 0x7f454c46** [comment]
If you wish, you can add this on top: https://github.com/0x7f454c46/criu/commit/0126dbeb21df3e32b094ebc0adbc7cc7e294a278

File: Makefile.config [hunk 2]
```diff
@@ -84,8 +84,10 @@ FEATURES_LIST	:= TCP_REPAIR STRLCPY STRLCAT PTRACE_PEEKSIGINFO \
 # $1 - config name
 define gen-feature-test
 ifeq ($$(call try-cc,$$(FEATURE_TEST_$(1)),$$(LIBS_FEATURES),$$(DEFINES)),true)
-	$(Q) echo '#define CONFIG_HAS_$(1)' >> $$@
+	echo '#define CONFIG_HAS_$(1)' >> $$@
 	$(Q) echo '' >> $$@
+else
+	echo '#undef CONFIG_HAS_$(1)' > /dev/null
```
Comment (avagin) [inline Makefile.config:90]: should it be something like this:
```
@@ -84,10 +84,10 @@ FEATURES_LIST       := TCP_REPAIR STRLCPY STRLCAT PTRACE_PEEKSIGINFO \
 # $1 - config name
 define gen-feature-test
 ifeq ($$(call try-cc,$$(FEATURE_TEST_$(1)),$$(LIBS_FEATURES),$$(DEFINES)),true)
-       echo '#define CONFIG_HAS_$(1)' >> $$@
+       $(E) 'CONFIG_HAS_$(1)=y'
        $(Q) echo '' >> $$@
 else
-       echo '#undef CONFIG_HAS_$(1)' > /dev/null
+       $(E) "#CONFIG_HAS_$(1) isn't set"
 endif
 endef
```

**Comment by avagin** [comment]
```
diff --git a/Makefile.config b/Makefile.config
index 9950134a4..721e5d2be 100644
--- a/Makefile.config
+++ b/Makefile.config
@@ -84,10 +84,9 @@ FEATURES_LIST        := TCP_REPAIR STRLCPY STRLCAT PTRACE_PEEKSIGINFO \
 # $1 - config name
 define gen-feature-test
 ifeq ($$(call try-cc,$$(FEATURE_TEST_$(1)),$$(LIBS_FEATURES),$$(DEFINES)),true)
-       echo '#define CONFIG_HAS_$(1)' >> $$@
-       $(Q) echo '' >> $$@
+       $(Q) echo '#define CONFIG_HAS_$(1)' >> $$@
 else
-       echo '#undef CONFIG_HAS_$(1)' > /dev/null
+       $(Q) echo '// CONFIG_HAS_$(1) is not set' >> $$@
 endif
 endef
 
avagin@avagin:~/git/checkpoint-restore/criu$ make V=1 include/common/config.h
Note: Building criu without amdgpu_plugin.
Note: libdrm and libdrm_amdgpu are required to build amdgpu_plugin.
Makefile.config:53: Warn: you have no libnftables installed
Makefile.config:54: Warn: Building without nftables support
touch .config
echo '#ifndef __CR_CONFIG_H__'				> include/common/config.h
echo '#define __CR_CONFIG_H__'				>> include/common/config.h
echo ''							>> include/common/config.h
echo '#define CONFIG_HAS_TCP_REPAIR' >> include/common/config.h
echo '#define CONFIG_HAS_STRLCPY' >> include/common/config.h
echo '#define CONFIG_HAS_STRLCAT' >> include/common/config.h
echo '// CONFIG_HAS_PTRACE_PEEKSIGINFO is not set' >> include/common/config.h
echo '#define CONFIG_HAS_SETPROCTITLE_INIT' >> include/common/config.h
echo '// CONFIG_HAS_MEMFD is not set' >> include/common/config.h
echo '#define CONFIG_HAS_TCP_REPAIR_WINDOW' >> include/common/config.h
echo '#define CONFIG_HAS_MEMFD_CREATE' >> include/common/config.h
echo '#define CONFIG_HAS_OPENAT2' >> include/common/config.h
echo '// CONFIG_HAS_NO_LIBC_RSEQ_DEFS is not set' >> include/common/config.h
cat .config | sed -n -e '/^[^#]/s/^/#define CONFIG_/p' >> include/common/config.h
echo '#endif /* __CR_CONFIG_H__ */'			>> include/common/config.h
avagin@avagin:~/git/checkpoint-restore/criu$ cat include/common/config.h
#ifndef __CR_CONFIG_H__
#define __CR_CONFIG_H__

#define CONFIG_HAS_TCP_REPAIR
#define CONFIG_HAS_STRLCPY
#define CONFIG_HAS_STRLCAT
// CONFIG_HAS_PTRACE_PEEKSIGINFO is not set
#define CONFIG_HAS_SETPROCTITLE_INIT
// CONFIG_HAS_MEMFD is not set
#define CONFIG_HAS_TCP_REPAIR_WINDOW
#define CONFIG_HAS_MEMFD_CREATE
#define CONFIG_HAS_OPENAT2
// CONFIG_HAS_NO_LIBC_RSEQ_DEFS is not set
#endif /* __CR_CONFIG_H__ */
```

File: Makefile.config (hunk 2 above)
Comment (osctobe) [reply Makefile.config:90]: I think not - this code generates a C header and the .config file is converted in another line in `config-header-rule` macro.

**Pull request merged**